src/main/java/com/conveyal/datatools/common/utils/Scheduler.java

```diff
@@ -1,63 +1,203 @@
+     * Cancels and removes all scheduled jobs for a given id and job class.
      */
-    private static void scheduleProjectAutoFetches() {
-        for (Project project : Persistence.projects.getAll()) {
-            if (project.autoFetchFeeds) {
-                ScheduledFuture scheduledFuture = ProjectController.scheduleAutoFeedFetch(project, 1);
-                autoFetchMap.put(project.id, scheduledFuture);
+    public static void removeJobsOfType(String id, Class<?> clazz, boolean mayInterruptIfRunning) {
+        int jobsCancelled = 0;
+        // First get the list of jobs belonging to the id (e.g., all jobs related to a feed source).
+        List<ScheduledJob> scheduledJobs = scheduledJobsById.get(id);
+        // Iterate over jobs, cancelling and removing only those matching the job class.
+        for (ScheduledJob scheduledJob : scheduledJobs) {
+            if (clazz.isInstance(scheduledJob.job)) {
+                scheduledJob.scheduledFuture.cancel(mayInterruptIfRunning);
+                scheduledJobsById.remove(id, scheduledJob);
+                jobsCancelled++;
             }
         }
+        // De-clutter the logs, but only logging when jobs have actually been cancelled/removed
+        if (jobsCancelled > 0) {
+            LOG.info("Cancelled/removed {} {} jobs for {}", clazz.getSimpleName(), jobsCancelled, id);
+        }
     }
 
     /**
-     * Schedule all feed version expiration notification jobs
+     * Schedule an action that fetches all the feeds in the given project according to the autoFetch fields of that project.
+     * Currently feeds are not auto-fetched independently, they must be all fetched together as part of a project.
+     * This method is called when a Project's auto-fetch settings are updated, and when the system starts up to populate
+     * the auto-fetch scheduler.
      */
-    private static void scheduleFeedNotificationExpirations() {
-        LOG.info("Scheduling feed expiration notifications");
+    public static void scheduleAutoFeedFetch (Project project, int intervalInDays) {
+        TimeUnit minutes = TimeUnit.MINUTES;
+        try {
+            // First cancel any already scheduled auto fetch task for this project id.
+            removeJobsOfType(project.id, FetchProjectFeedsJob.class, true);
 
-        // get all active feed sources
-        for (FeedSource feedSource : Persistence.feedSources.getAll()) {
-            // schedule expiration notification jobs for the latest feed version
-            feedSource.scheduleExpirationNotifications();
+            ZoneId timezone = getTimezone(project.defaultTimeZone);
+            LOG.info("Scheduling auto-fetch for projectID: {}", project.id);
+
+            // NOW in default timezone
+            ZonedDateTime now = ZonedDateTime.ofInstant(Instant.now(), timezone);
+
+            // Scheduled start time
+            ZonedDateTime startTime = LocalDateTime.of(LocalDate.now(),
```

Comment:
Can you format this a little better?  Maybe add a linebreak after the `of(` and make the ending paren on a newline.

Reply:
Yea, sorry about that. I think the auto-formatter got a hold of it.